scripts/compile_sass.py

```diff
@@ -0,0 +1,436 @@
+#!/usr/bin/env python
+"""
+Defines a CLI for compiling Sass (both default and themed) into CSS.
+
+Should be run from the root of edx-platform using `npm run` wrapper.
+Requirements for this scripts are stored in requirements/edx/assets.in.
+
+Get more details:
+
+     npm run compile-sass -- --help
+     npm run compile-sass -- --dry
+
+Setup (Tutor and Devstack will do this for you):
+
+    python -m venv venv
+    . venv/bin/activate
+    pip install -r requirements/edx/assets.txt
+
+Usage:
+
+     npm run compile-sass             # prod, no args
+     npm run compile-sass -- ARGS     # prod, with args
+     npm run compile-sass-dev         # dev, no args
+     npm run compile-sass-dev -- ARGS # dev, with args
+
+This script is intentionally implemented in a very simplistic way. It prefers repetition
+over abstraction, and its dependencies are minimal (just click and libsass-python, ideally).
+We do this because:
+
+* If and when we migrate from libsass-python to something less ancient like node-sass or dart-sass,
+  we will want to re-write this script in Bash or JavaScript so that it can work without any
+  backend tooling. By keeping the script dead simple, that will be easier.
+
+* The features this script supports (legacy frontends & comprehensive theming) are on the way out,
+  in favor of micro-frontends, branding, and Paragon design tokens. We're not sure how XBlock
+  view styling will fit into that, but it probably can be much simpler than comprehensive theming.
+  So, we don't need this script to be modular and extensible. We just need it to be obvious, robust,
+  and easy to maintain until we can delete it.
+
+See docs/decisions/0017-reimplement-asset-processing.rst for more details.
+"""
+from __future__ import annotations
+
+import glob
+import subprocess
+from pathlib import Path
+
+import click
+
+
+# Accept both long- and short-forms of these words, but normalize to long form.
+# We accept both because edx-platform asset build scripts historically use the short form,
+# but NODE_ENV uses the long form, so to make them integrate more seamlessly we accept both.
+NORMALIZED_ENVS = {
+    "prod": "production",
+    "dev": "development",
+    "production": "production",
+    "development": "development",
+}
+
+
+@click.option(
+    "-T",
+    "--theme-dir",
+    "theme_dirs",
+    metavar="PATH",
+    multiple=True,
+    envvar="EDX_PLATFORM_THEME_DIRS",
```

Comment:
This is a new env var, and is mentioned in the ADR. Eventually, I will document it in the edx-platform README.